rpc/client.go

```diff
@@ -39,15 +40,29 @@ type response[T serde.ResponsePayload] struct {
 }
 
 // NewRpcClient creates a new RpcClient
-func NewRpcClient(rpcServerUrl string, myAddress types.Address, logger zerolog.Logger, trans transport.Requester) (*RpcClient, error) {
-	c := &RpcClient{trans, myAddress, logger, &safesync.Map[chan struct{}]{}, &safesync.Map[chan query.LedgerChannelInfo]{}, &safesync.Map[chan query.PaymentChannelInfo]{}}
+func NewRpcClient(rpcServerUrl string, logger zerolog.Logger, trans transport.Requester) (*RpcClient, error) {
+	c := &RpcClient{trans, logger, &safesync.Map[chan struct{}]{}, &safesync.Map[chan query.LedgerChannelInfo]{}, &safesync.Map[chan query.PaymentChannelInfo]{}}
 	err := c.subscribeToNotifications()
 	if err != nil {
 		return nil, err
 	}
 	return c, nil
 }
 
+// NewHttpRpcClient creates a new RpcClient using an http transport
+func NewHttpRpcClient(rpcServerUrl string) (*RpcClient, error) {
+	transport, err := ws.NewWebSocketTransportAsClient(rpcServerUrl)
+	if err != nil {
+		return nil, err
+	}
+	c := &RpcClient{transport, zerolog.New(os.Stdout), &safesync.Map[chan struct{}]{}, &safesync.Map[chan query.LedgerChannelInfo]{}, &safesync.Map[chan query.PaymentChannelInfo]{}}
```

Comment:
As a temporary measure, I wanted to remove the need to pass a `zerolog.Logger` into the constructor.